internal/document/frontmatter.go

```diff
@@ -67,10 +87,104 @@ func ParseFrontmatter(raw string) (f Frontmatter, info FrontmatterParseInfo) {
 	return
 }
 
+func ToJSONStr(dest map[string]interface{}, bytes []byte) string {
```

Comment:
General comments to this set of functions:
* They are not used outside of `internal/document`; they should be unexported.
* It seems that it's a quite serious bug if unmarshaling fails; I think we should report that and handle, otherwise it will be extremely hard to find the bug. It seems that this function is only called in `Deserialize()` which returns an error.